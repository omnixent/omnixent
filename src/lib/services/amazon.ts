import { ServiceArgs } from '.';
import { normalizeServiceArgs } from '../utils';

const endpoint = 'https://completion.amazon.co.uk/api/2017/suggestions';

export function formatURI(options: ServiceArgs): string {
  const { t, hl } = normalizeServiceArgs(options);
  return `${endpoint}?page-type=Gateway&client-info=amazon-search-ui&mid=APJ6JRA9NG5V4&alias=aps&suggestion-type=KEYWORD&prefix=${t}&lop=${hl}`;
}

export function extractBody(response: string) {
  return JSON?.parse(response)?.suggestions?.map(({ value }: any) => value);
}
