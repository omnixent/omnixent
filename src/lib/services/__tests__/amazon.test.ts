import { formatURI, extractBody } from '../amazon';

test('Testing google formatURI', () => {
  expect(formatURI({ term: 'Node.js', language: 'it', country: 'us' })).toMatchInlineSnapshot(
    `"https://completion.amazon.co.uk/api/2017/suggestions?page-type=Gateway&client-info=amazon-search-ui&mid=APJ6JRA9NG5V4&alias=aps&suggestion-type=KEYWORD&prefix=node.js&lop=us-IT"`,
  );
  expect(
    formatURI({ term: 'blue jeans brands', language: 'en', country: 'de' }),
  ).toMatchInlineSnapshot(
    `"https://completion.amazon.co.uk/api/2017/suggestions?page-type=Gateway&client-info=amazon-search-ui&mid=APJ6JRA9NG5V4&alias=aps&suggestion-type=KEYWORD&prefix=blue%20jeans%20brands&lop=de-EN"`,
  );
});

test('Testing google extractBody', () => {
  expect(
    extractBody(
      `{"alias":"aps","prefix":"node.js","suffix":null,"suggestions":[{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"node.js","refTag":"nb_sb_ss_i_1_7","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"blackListed":false,"xcatOnly":false,"spellCorrected":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"node.js design patterns","refTag":"nb_sb_ss_i_2_7","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"blackListed":false,"xcatOnly":false,"spellCorrected":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"node.js italiano","refTag":"nb_sb_ss_i_3_7","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"blackListed":false,"xcatOnly":false,"spellCorrected":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"node.js. guida completa per lo sviluppatore","refTag":"nb_sb_ss_i_4_7","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"blackListed":false,"xcatOnly":false,"spellCorrected":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"node.js o'reilly","refTag":"nb_sb_ss_i_5_7","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"blackListed":false,"xcatOnly":false,"spellCorrected":false}],"suggestionTitleId":null,"responseId":"EIDBZJUJJZQ1","shuffled":false}`,
    ),
  ).toMatchSnapshot();
  expect(
    extractBody(
      `{"alias":"aps","prefix":"blue jeans","suffix":null,"suggestions":[{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"versace blue jeans profumo uomo","refTag":"nb_sb_ss_i_1_10","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"spellCorrected":false,"blackListed":false,"xcatOnly":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"blue jeans uomo","refTag":"nb_sb_ss_i_2_10","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"spellCorrected":false,"blackListed":false,"xcatOnly":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"blue jeans","refTag":"nb_sb_ss_i_3_10","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"spellCorrected":false,"blackListed":false,"xcatOnly":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"blue jeans versace","refTag":"nb_sb_ss_i_4_10","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"spellCorrected":false,"blackListed":false,"xcatOnly":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"blue jeans franco126","refTag":"nb_sb_ss_i_5_10","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"spellCorrected":false,"blackListed":false,"xcatOnly":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"blue jeans donna","refTag":"nb_sb_ss_i_6_10","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"spellCorrected":false,"blackListed":false,"xcatOnly":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"blue jeans libri","refTag":"nb_sb_ss_i_7_10","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"spellCorrected":false,"blackListed":false,"xcatOnly":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"blue jeans franco126 vinile","refTag":"nb_sb_ss_i_8_10","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"spellCorrected":false,"blackListed":false,"xcatOnly":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"blue jeans cable","refTag":"nb_sb_ss_i_9_10","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"spellCorrected":false,"blackListed":false,"xcatOnly":false},{"suggType":"KeywordSuggestion","type":"KEYWORD","value":"blue jeans vinile","refTag":"nb_sb_ss_i_10_10","strategyId":"organic","ghost":false,"help":false,"queryUnderstandingFeatures":[{"source":"QU_TOOL","annotations":[]}],"fallback":false,"spellCorrected":false,"blackListed":false,"xcatOnly":false}],"suggestionTitleId":null,"responseId":"34H2MBKTAESN9","shuffled":false}`,
    ),
  ).toMatchSnapshot();
});
