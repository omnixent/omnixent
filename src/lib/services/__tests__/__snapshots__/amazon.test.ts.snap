// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing google extractBody 1`] = `
Array [
  "node.js",
  "node.js design patterns",
  "node.js italiano",
  "node.js. guida completa per lo sviluppatore",
  "node.js o'reilly",
]
`;

exports[`Testing google extractBody 2`] = `
Array [
  "versace blue jeans profumo uomo",
  "blue jeans uomo",
  "blue jeans",
  "blue jeans versace",
  "blue jeans franco126",
  "blue jeans donna",
  "blue jeans libri",
  "blue jeans franco126 vinile",
  "blue jeans cable",
  "blue jeans vinile",
]
`;
